feat(http): http requests to implement xhr timeout (#36681)
angular/angular: aio/content/guide/http.md

A raw error object is not particularly useful as feedback.
In addition to detecting that an error has occurred, you need to get error details and use those details to compose a user-friendly response.

Three types of errors can occur.

* The server backend might reject the request, returning an HTTP response with a status code such as 404 or 500. These are error _responses_.

* Something could go wrong on the client-side such as a network error that prevents the request from completing successfully or an exception thrown in an RxJS operator. These errors produce JavaScript `ErrorEvent` objects.

> * The server took too long to respond. The timeout can be passed in as an optional parameter with the request object or the `HttpClient`'s request methods.

Reviewer: Maybe it is worth mentioning that this can only happen if you specified a timeout.

> * The server took too long to respond, but only when a timeout value has been passed in as an optional parameter with the request object or the `HttpClient`'s request methods.

Reviewer 2: Perhaps more concise: 
```suggestion
* The server took too long to respond. This can occur only when the request specifies an explicit `timeout`.
```
